typing/ctype.ml

```diff
@@ -2585,13 +2585,7 @@ let add_gadt_equation uenv source destination =
 
   let env = get_env uenv in
 
-  let staging_level = Env.get_env_level env in
-  let staging_mode = Env.get_env_mode env in
-
-  if (* staging_level <> 0 ||*) staging_mode <> M_C then
-    Location.prerr_warning Location.none (Warnings.Maco_dev 
-      ("avoided gadt constraint at level "^(string_of_int staging_level)))
-  else begin
+  begin
```

Comment:
I think this removes too much: we don't need the warning, but we probably do still need the check `staging_mode <> M_C` that guards the `else` branch.